### PR TITLE
chore: [DX-2714] remove core sdk as a dependency

### PR DESCRIPTION
**Pull request by carleeto**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

# Summary
Remove Core SDK as a dependency

# Anything else worth calling out?
No behavioural code changes

### CODE REVIEW

File: tests/func-tests/imx/step-definitions/burning.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   ImxModuleConfiguration,
   GenericIMXProvider,
   ProviderConfiguration,
+  UnsignedTransferRequest,
 } from '@imtbl/sdk/x';
 import { configuration, StepSharedState } from './stepSharedState';
 
 export class Burning {
-  constructor(protected stepSharedState: StepSharedState) {}
+  constructor(protected stepSharedState: StepSharedState) { }
```
Comment (wdinau) [inline tests/func-tests/imx/step-definitions/burning.ts:11]: is this normal? linting?

File: tests/func-tests/imx/step-definitions/minting.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+import {
+  IMXClient, ImxModuleConfiguration, MintFee, MintTokenDataV2,
+} from '@imtbl/sdk/x';
 import { env, repeatCheck300 } from '../common';
 import { configuration, StepSharedState } from './stepSharedState';
 
 // @binding([StepSharedState])
 export class Minting {
-  constructor(protected stepSharedState: StepSharedState) {}
+  constructor(protected stepSharedState: StepSharedState) { }
```
Comment (wdinau) [inline tests/func-tests/imx/step-definitions/minting.ts:11]: same here?

**Review by wdinau — APPROVED** [review]

File: tests/func-tests/imx/step-definitions/burning.ts (hunk 1 above)
Comment (carleeto) [reply tests/func-tests/imx/step-definitions/burning.ts:11]: There's definitely a problem with eslint and prettier configs...I tried fixing them, but that introduced more issues, so I reverted the config changes. This one got through the cracks.

File: tests/func-tests/imx/step-definitions/minting.ts (hunk 2 above)
Comment (carleeto) [reply tests/func-tests/imx/step-definitions/minting.ts:11]: yeah...same

**Review by wdinau — APPROVED** [review]

**Review by nattb8 — APPROVED** [review]

**Review by ckailash — APPROVED** [review]

**Pull request merged**